### PR TITLE
Ensure backwards compatability of new time_resolved key with previous stores

### LINKED ISSUE #989: Ensure previous keyword can still be used to extract high time resolution footprints
PR #968 updated the default keyword for co2 footprints to be `time_resolved` rather than the previous `high_time_resolution`. All object stores created after this update will include the new keyword.

However, as this changes was incorporated after the breaking 0.8.0 release, we need to ensure backwards compatibility so that for any object stores created prior to this change, it is still possible to extract footprints labelled within `high_time_resolution` rather than `time_resolved`.

### PR DESCRIPTION
**Pull request by rt17603**
* **Summary of changes** (Bug fix, feature, docs update, ...)

Updates to search to ensure backwards compatibility with object stores created prior to changes to keywords in PR #968 when retrieving time resolved footprints. Previous PR #968 was to improve the interface by introducing the new, and more accurate, `time_resolved` keyword in favour of `high_time_resolution` but for now we still need to be able retrieve appropriate footprints which are only labelled with `high_time_resolution` and NOT `time_resolved`.

These updates include:
 - Update to `openghg.retrieve.search` (actually `_base_search`) to allow dictionary inputs to be passed so that multiple options for keywords can be used (expands on current ability to do this with lists). This performs an OR search with the other keywords.
 - Update to `openghg.retrieve.search_footprints` to create an appropriate input for this with the two possible keywords
 - Tests to check both keywords and to mimic previous setup to ensure the footprint with the previous labeling can be retrieved

*Note: solution here is pretty rough and ready and test is a bit scrappy - happy for alternative suggestions for how to approach this but do want to get a fix in asap. See #989 for alternative considered to update in search interface (use a dataclass rather than a dict).*

* **Please check if the PR fulfills these requirements**

- [x] Closes #989
- [x] [Tests added and passed](https://docs.openghg.org/development/python_devel.html#testing) if fixing a bug or adding a new feature
- [x] All code checks passing - `black --line-length 110` run over code, `mypy` and `flake8` not showing any errors
- [x] Documentation and tutorials updated/added
- [x] Added an entry in the latest `CHANGELOG.md` file if fixing a bug or adding a new feature
- ~Added any new requirements to `requirements.txt` and `recipes/meta.yaml`~

### CODE REVIEW

**Comment by rt17603** [comment]
Seem to be getting a `mypy` error in the CI that is unrelated to the changes:
```
openghg/util/_util.py:103: error: Need type annotation for "matches"  [var-annotated]
openghg/util/_util.py:135: error: Need type annotation for "matches"  [var-annotated]
Found 2 errors in 1 file (checked 111 source files)
```

I'm not currently able to replicate this locally.

**Comment by rt17603** [comment]
Fix put in for failing tests in #991 - this branch has been merged into this PR to make sure this can pass the CI as well.

**Comment by rt17603** [comment]
I'm currently trying to add some small updates to the documentation in relation to this PR but I'm having some issues getting the tutorial code to run. So I'm going to leave the "Documentation and tutorials updated/added" box unticked and the rest is ready for review.

**Comment by rt17603** [comment]
Small changes to the documentation have now been added.

Merged in `BUGFIX_zarr_store_check` to this branch which is why the checks are now failing (just a formatting issue). This should be fixed when `BUGFIX_zarr_store_check` is updated, merged into devel and devel can be merged into this PR.

File: doc/source/tutorials/local/Exploring_data/Searching_and_plotting.rst [hunk 1]
```diff
@@ -85,11 +85,33 @@ to extract, for example, just the methane data:
...
+.. code:: ipython3
+
+    from openghg.retrieve import search_surface
+
+    tac_surface_search = search_surface(site="TAC", species="ch4", inlet=["100m", "185m"])
+    tac_surface_search.results    
+
+This will return results from both the 100m and 185m inlets (but not the 54m inlet).
+
+Note: it is also possible to specify a dictionary to provide an option between
```
Comment (gareth-j) [inline doc/source/tutorials/local/Exploring_data/Searching_and_plotting.rst:105]: Do we need to mention an option that isn't covered by the tutorial?

**Review by gareth-j — APPROVED** [review]
Looks good!

File: doc/source/tutorials/local/Exploring_data/Searching_and_plotting.rst (hunk 1 above)
Comment (rt17603) [reply doc/source/tutorials/local/Exploring_data/Searching_and_plotting.rst:105]: Is there a downside to mentioning it? Mentioning it exists at least means if someone has an alternative use for it that wasn't previously considered they know it is there as an option.

**Review by SutarPrasad — APPROVED** [review]

File: doc/source/tutorials/local/Exploring_data/Searching_and_plotting.rst (hunk 1 above)
Comment (gareth-j) [reply doc/source/tutorials/local/Exploring_data/Searching_and_plotting.rst:105]: Yeah that's cool.

**Review by gareth-j — APPROVED** [review]
Looks good!

**Pull request merged**